### PR TITLE
Adding Telford and Wrekin Council UK

### PR DESCRIPTION
**Pull request by alistairclare**
Adding Telford.gov.uk source. Documentation .md file created for telford_gov_uk detailing the configuration and collection types. Spotted error with special characters in the info.md and readme.md that have been fixed.

Hope its all ok - first contribution and first attempt a Home Assistant / Python change / Github :)

Tests run and completed with results below:

python ./test_sources.py -s telford_gov_uk -i -l
Testing source telford_gov_uk ...
found 4 entries for 10 Long Row Drive, Lawley
2023-01-11: Recycling (Purple/Blue) [mdi:recycle]
2023-01-18: General (Red) [mdi:trash-can]
2023-01-11: Garden (Green) [mdi:leaf]
2023-01-11: Food (Silver) [mdi:food]
found 4 entries for 126 Dunsheath, Telford
2023-01-06: Recycling (Purple/Blue) [mdi:recycle]
2023-01-12: General (Red) [mdi:trash-can]
2023-01-06: Garden (Green) [mdi:leaf]
2023-01-06: Food (Silver) [mdi:food]
found 4 entries for 11 Pinewoods, Telford
2023-01-10: Recycling (Purple/Blue) [mdi:recycle]
2023-01-10: Garden (Green) [mdi:leaf]
2023-01-10: Food (Silver) [mdi:food]
2023-01-17: General (Red) [mdi:trash-can]

### CODE REVIEW

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/telford_gov_uk.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    "11 Pinewoods, Telford": {"post_code": "TF10 9LN", "name_number": "11"}
+}
+
+API_URLS = {
+    "address_search": "https://dac.telford.gov.uk/BinDayFinder/Find/PostcodeSearch?postcode={postcode}",
+    "collection": "https://dac.telford.gov.uk/BinDayFinder/Find/PropertySearch?uprn={uprn}",
+}
+
+# Map the names to something readable
+NAMES = {
```
[the remark this answers is not in the document]
Comment (alistairclare) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/telford_gov_uk.py]: Latest commit has removed the type mapping.

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/telford_gov_uk.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ 
+    date = datetime.datetime.strptime(datestring, "%A %d %B").date()
+
+    # Calculate the year. As we only get collections 2 weeks in advance we can assume the current
+    # year unless the month is January in December where it will be next year
+
+    if (date.month == 1) and (today.month == 12):
+        year=year+1
+
+    date = date.replace(year=year)
```
[the remark this answers is not in the document]
Comment (alistairclare) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/telford_gov_uk.py]: date parsing moved into the if statement

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/telford_gov_uk.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                    self._uprn=property['UPRN']
+
+            if not self._uprn:
+                raise Exception(
+                    f"Could not find address {self._post_code} {self._name_number}")
+        
+        # Get the collection information
+
+        q = str(API_URLS["collection"]).format(
+            uprn=self._uprn
```
[the remark this answers is not in the document]
Comment (alistairclare) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/telford_gov_uk.py]: Updated to use params argument

**Comment by alistairclare** [comment]
Changes made and tests still pass. Documentation updated to reflect the removal of the type mapping. Hope this is ok now.

Testing source telford_gov_uk ...
  found 4 entries for 10 Long Row Drive, Lawley
    2023-01-11: Purple / Blue Containers [mdi:recycle]
    2023-01-18: Red Top Container [mdi:trash-can]
    2023-01-11: Green Container [mdi:leaf]
    2023-01-11: Silver Containers [mdi:food]
  found 4 entries for 126 Dunsheath, Telford
    2023-01-06: Purple / Blue Containers [mdi:recycle]
    2023-01-12: Red Top Container [mdi:trash-can]
    2023-01-06: Green Container [mdi:leaf]
    2023-01-06: Silver Containers [mdi:food]
  found 4 entries for 11 Pinewoods, Telford
    2023-01-10: Purple / Blue Containers [mdi:recycle]
    2023-01-10: Green Container [mdi:leaf]
    2023-01-10: Silver Containers [mdi:food]
    2023-01-17: Red Top Container [mdi:trash-can]

**Comment by mampfes** [comment]
Thanks for your contribution.

**Pull request merged**